### ISSUE TITLE
`explicit_acl` false positive with static subscript function

### ISSUE THREAD

**Issue by jakecast** [realm/SwiftLint#5457]
### New Issue Checklist

- [✅] Updated SwiftLint to the latest version
- [✅] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`explicit_acl` is reporting a false positive violation with a `static subscript` function in Swiftlint 0.54.0

<img width="1154" alt="Screenshot 2024-02-06 at 10 34 31" src="https://github.com/realm/SwiftLint/assets/868863/7ed2b006-2597-4920-9668-73e78ae023d7">

Causes warning:
```swift
extension NSObject {
    internal static subscript(name: String) -> NSObject? {
        nil
    }
}
```

```swift
extension NSObject {
    internal class subscript(name: String) -> NSObject? {
        nil
    }
}
```

Does not cause warning:
```swift
extension NSObject {
    static internal subscript(name: String) -> NSObject? {
        nil
    }
}
```

```swift
extension NSObject {
    class internal subscript(name: String) -> NSObject? {
        nil
    }
}
```

**Comment by SimplyDanny** [comment]
Seems like with the recent rewrite in #5382 this can be considered fixed. Please check with `main` to make sure.
